### PR TITLE
Feature/rework worker pool

### PR DESCRIPTION
**Pull request by mosquito**

### CODE REVIEW

File: aiomisc/compat.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+def _get_running_loop_backport() -> asyncio.AbstractEventLoop:
+    loop = asyncio.get_event_loop()
+    if loop.is_running():
+        return loop
+    raise RuntimeError("no running event loop")
+
+
+get_running_loop = getattr(
+    asyncio, "get_running_loop", _get_running_loop_backport,
+)
```
Comment (dizballanze) [inline aiomisc/compat.py:52]: Maybe we should drop python3.6 support? It already reached end-of-life.

File: aiomisc_worker/forking.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    atexit.register(at_exit)
+
+    log.debug("Forking %d processes", len(worker_ids))
+    for worker_id in worker_ids:
+        fork(worker_id, cookie, address)
+
+    log.debug("Waiting workers")
+    while not STOPPING.is_set():
+        pid, status = os.wait()
+        log.debug("Worker PID: %d exited with status %d", pid, status)
```
Comment (dizballanze) [inline aiomisc_worker/forking.py:107]: warning maybe?

File: tests/test_worker_pool.py [hunk 3]
```diff
@@ -75,13 +79,28 @@ async def test_incomplete_task_kill(worker_pool):
...
+                stats[metric.metric] = metric.value
+        return stats
+
+    stats = get_stats()
+    try:
+        while stats["spawning"] < PROCESS_NUM:
+            await asyncio.sleep(0.1)
+            stats = get_stats()
+    except asyncio.CancelledError:
+        breakpoint()
```
Comment (dizballanze) [inline tests/test_worker_pool.py:101]: ?

File: aiomisc/compat.py (hunk 1 above)
Comment (mosquito) [reply aiomisc/compat.py:52]: it's unfortunately most popular python in 2021

File: aiomisc_worker/forking.py (hunk 2 above)
Comment (mosquito) [reply aiomisc_worker/forking.py:107]: might be warning when not zero exit code

File: tests/test_worker_pool.py (hunk 3 above)
Comment (mosquito) [reply tests/test_worker_pool.py:101]: oops

**Pull request merged**